### PR TITLE
JSON-RPC API Updates for 2.0

### LINKED ISSUE #1317: Document New JSON-RPC API Endpoints
[Casper Node PR#4351](https://github.com/casper-network/casper-node/pull/4351) added the following endpoints:

- `account_put_transaction`
- `info_get_transaction`
- `speculative_exec_txn`

These will need to be added to our [JSON-RPCI API documentation](https://docs.casper.network/developers/json-rpc/)

### LINKED ISSUE #1342: Update `state_get_account_info` in the JSON-RPC Docs
`state_get_account_info` now takes in `account_identifier` rather than `public_key`, so that either a Public Key or an Account Hash can be used.

The docs need to be updated to reflect this.

### PR DESCRIPTION
**Pull request by ACStoneCL**
### What does this PR fix/introduce?
This PR grew out of control from the Account/Contract merge ticket, but covers many items we would need to finish for 2.0.

It includes all (current) new endpoints for the JSON-RPC and associated types, as well as some of the account/contract merge - but I will finish that in another PR.

Closes #1342
Closes #1317

### Additional context
[Document New JSON-RPC API Endpoints #1317](https://github.com/casper-network/docs/issues/1317)
[Update state_get_account_info in the JSON-RPC Docs #1342](https://github.com/casper-network/docs/issues/1342)

### Checklist
(Delete any that aren't relevant)

- [x] Docs are successfully building - `yarn install && yarn run build`.
- [x] For new **internal** links I used *relative file paths* (with .md extension) - e.g. `../../faq/faq-general.md` - instead of introducing *absolute file path*, or *relative/absolute URL*.
- [x] All external links have been verified with `yarn run check:externals`.
- [x] My changes follow the [Casper docs style guidelines](https://docs.casper.network/resources/contribute-to-docs/).
- [x] All technical procedures have been tested (if you want help with this, mention it in [Reviewers](#reviewers)).

### Reviewers
// Tag whoever needs to review. If you're not sure, leave blank.

### CODE REVIEW

**Comment by ACStoneCL** [comment]
Still trying to figure out these new build failures, might have to reach out to a dev.

**Comment by ipopescu** [comment]
Taking a look. The fix for the build is here: https://github.com/casper-network/docs/pull/1339/commits/81d370addee60e43047dbeee8c33cea2e333676e

File: source/docs/casper/developers/json-rpc/types_chain.md [hunk 1]
```diff
@@ -466,7 +842,19 @@ Required Parameters:
...
 ## ExecutionResult {#executionresult}
 
-The result of executing a single Deploy.
+The versioned result of executing a single Deploy.
+
+One of:
+
+* [`Version1`](#executionresultv1) Version 1 of execution result type.
+
+* [`Version2`](#executionresultv2) VErsion 2 of execution result type.
```
Comment (ipopescu) [inline source/docs/casper/developers/json-rpc/types_chain.md:851]:
```suggestion
* [`Version2`](#executionresultv2) Version 2 of execution result type.
```

File: source/docs/casper/developers/json-rpc/types_chain.md [hunk 2]
```diff
@@ -342,9 +670,35 @@ Enum describing the possible access control options for a contract entry point.
...
 Context of an entry point execution.
 
-* `session` Executes in the caller's context.
+* `Session` Runs as session code in the context of the caller. Deprecated and retained to allow read back of legacy stored session.
+
+* `AddressableEntity` Runs within called entity's context.
+
+* `Factory` This entry point is intended to extract a subset of bytecode. Runs within called entity's context.
+
+## EraEndV1 {#eraendv1}
```
Comment (ipopescu) [inline source/docs/casper/developers/json-rpc/types_chain.md:679]: Would it help to add a note somewhere pointing out that V1 and Version 1 (below) refer to all the Casper versions up to 2.0?

File: source/docs/casper/developers/json-rpc/types_chain.md [hunk 3]
```diff
@@ -490,6 +878,34 @@ The result of executing a single Deploy.
...
+
+    Required Parameters:
+
+    [`effects`](#effects)
+
+    [`transfers`](#transferaddr)
+
+    [`cost`](#u512)
+
+    `error_message` The error message associated with executing the Deploy.
```
Comment (ipopescu) [inline source/docs/casper/developers/json-rpc/types_chain.md:897]: For consistency with the definition above:
```suggestion
    `error_message` The error message associated with executing the deploy.
```

File: source/docs/casper/developers/json-rpc/types_chain.md [hunk 4]
```diff
@@ -500,6 +916,8 @@ Identifier for possible ways to query global state.
 
 * [`BlockHash`](#blockhash) Query using a Block hash.
```
Comment (ipopescu) [inline source/docs/casper/developers/json-rpc/types_chain.md:917]: For consistency with the line below:
```suggestion
* [`BlockHash`](#blockhash) Query using a block hash.
```

File: source/docs/casper/developers/json-rpc/types_chain.md [hunk 5]
```diff
@@ -874,6 +1490,210 @@ Human-readable duration.
...
+
+One of:
+
+* `Custom` A non-native, arbitrary entry point.
+
+* `Transfer` The `transfer` native entry point, used to reference motes from a source purse to a target purse.
+
+* `AddBid` The `add_bid` native entry point, used to create or top off a bid purse.
+
+* `WithdrawBid` The `withdraw_bid` native entry point, used to decrease a stake.
```
Comment (ipopescu) [inline source/docs/casper/developers/json-rpc/types_chain.md:1521]: Could we add this detail:
```suggestion
* `WithdrawBid` The `withdraw_bid` native entry point, used to decrease a validator's stake.
```

File: source/docs/casper/developers/json-rpc/types_chain.md [hunk 6]
```diff
@@ -874,6 +1490,210 @@ Human-readable duration.
...
+
+* [`Version`](#transactionv1hash)
+
+## TransactionInvocationTarget {#transactioninvocationtarget}
+
+The identifier of a `stored` transaction target.
+
+One of:
+
+* `InvocableEntity` The hex-encoded entity address indetifying the invocable entity.
```
Comment (ipopescu) [inline source/docs/casper/developers/json-rpc/types_chain.md:1545]:
```suggestion
* `InvocableEntity` The hex-encoded entity address identifying the invocable entity.
```

File: source/docs/casper/developers/json-rpc/types_chain.md [hunk 7]
```diff
@@ -50,6 +86,50 @@ Required Parameters:
...
+
+An array of [NamedKeys](#namedkey).
+
+## Array_of_NamedUserGroup {#array-of-namedusergroup}
+
+An array of [NamedUserGroups](#namedusergroup).
+
+## Array_of_PublicKeyAndBid {#array-of-publickeyandbid}
+
+An array of [bids associated with give public keys](#publickeyandbid).
```
Comment (ipopescu) [inline source/docs/casper/developers/json-rpc/types_chain.md:123]:
```suggestion
An array of [bids associated with given public keys](#publickeyandbid).
```

File: source/docs/casper/developers/json-rpc/types_chain.md [hunk 8]
```diff
@@ -102,10 +186,128 @@ Additional Parameters:
 
 * [`vesting_schedule`](#vestingschedule) Vesting schedule for a genesis validator. `None` if non-genesis validator.
 
+## BidKind {#bidkind}
+
+Auction bid variants.
+
+One of:
+
+* `Unified` A unified record indexed on validator data, with an embedded collection of all delegator bids assigned to that validator. The `Unified`` variant is for legacy retrograde support, new instances will not be created going forward.
```
Comment (ipopescu) [inline source/docs/casper/developers/json-rpc/types_chain.md:195]:
```suggestion
* `Unified` A unified record indexed on validator data, with an embedded collection of all delegator bids assigned to that validator. The `Unified`` variant is for legacy retrograde support; new instances will not be created going forward.
```

File: source/docs/casper/developers/json-rpc/types_chain.md [hunk 9]
```diff
@@ -114,25 +316,93 @@ Identifier for possible ways to retrieve a Block.
...
+
+* [`block_hash`](#blockhash)
+
+Additional Parameters:
+
+* `block_height` The height of the block, if known.
+
+## BlockTime {#blocktime}
+
+A newtype wrapping a `u64` which represents the block time.
```
Comment (ipopescu) [inline source/docs/casper/developers/json-rpc/types_chain.md:353]:
```suggestion
A newtype wrapping a `u64`, which represents the block time.
```

File: source/docs/casper/developers/json-rpc/types_chain.md [hunk 10]
```diff
@@ -114,25 +316,93 @@ Identifier for possible ways to retrieve a Block.
...
+
+## Bytes {#bytes}
+
+Hex-encoded bytes.
 
-    * [`block_hash`](#blockhash-blockhash) The block hash.
+## ByteCode {#bytecode}
 
-    * `block_height` The height of the block, if known.
+A container for contract's Wasm bytes.
```
Comment (ipopescu) [inline source/docs/casper/developers/json-rpc/types_chain.md:385]:
```suggestion
A container for a contract's Wasm bytes.
```

File: source/docs/casper/developers/json-rpc/types_chain.md [hunk 11]
```diff
@@ -342,9 +670,35 @@ Enum describing the possible access control options for a contract entry point.
 
 Context of an entry point execution.
 
-* `session` Executes in the caller's context.
+* `Session` Runs as session code in the context of the caller. Deprecated and retained to allow read back of legacy stored session.
+
+* `AddressableEntity` Runs within called entity's context.
```
Comment (ipopescu) [inline source/docs/casper/developers/json-rpc/types_chain.md:675]:
```suggestion
* `AddressableEntity` Runs within the called entity's context.
```

File: source/docs/casper/developers/json-rpc/types_chain.md [hunk 12]
```diff
@@ -342,9 +670,35 @@ Enum describing the possible access control options for a contract entry point.
...
+
+* `Factory` This entry point is intended to extract a subset of bytecode. Runs within called entity's context.
```
Comment (ipopescu) [inline source/docs/casper/developers/json-rpc/types_chain.md:677]:
```suggestion
* `Factory` This entry point is intended to extract a subset of bytecode. Runs within the called entity's context.
```

File: source/docs/casper/developers/json-rpc/types_chain.md [hunk 13]
```diff
@@ -466,7 +842,19 @@ Required Parameters:
...
+
+* [`Version2`](#executionresultv2) VErsion 2 of execution result type.
+
+## ExecutionResultV1 {#executionresultV1}
+
+The result of executing a single deploy.
+
+One of:
 
 * `Failure` The result of a failed execution`
```
Comment (ipopescu) [inline source/docs/casper/developers/json-rpc/types_chain.md:859]:
```suggestion
* `Failure` The result of a failed execution
```

File: source/docs/casper/developers/json-rpc/types_chain.md [hunk 14]
```diff
@@ -684,10 +1124,38 @@ Required Parameters:
 
 * [`weight`](#u512)
 
+## Key {#key}
+
+The key as a formatted string, under which data can be stored in global state.
+
 ## Merkle_Proof {#merkle-proof}
 
 A merkle proof is a construction created using a merkle trie that allows verification of the associated hashes.
```
Comment (ipopescu) [inline source/docs/casper/developers/json-rpc/types_chain.md:1133]:
```suggestion
A Merkle proof is a construction created using a Merkle trie that allows verification of the associated hashes.
```

File: source/docs/casper/developers/json-rpc/types_chain.md [hunk 15]
```diff
@@ -684,10 +1124,38 @@ Required Parameters:
...
 
+## MessageChecksum {#messagechecksum}
+
+Message checksum as a formatted string.
+
+## MessageTopic {#messagetopic}
+
+A topic for contract level messages.
```
Comment (ipopescu) [inline source/docs/casper/developers/json-rpc/types_chain.md:1141]:
```suggestion
A topic for contract-level messages.
```

File: source/docs/casper/developers/json-rpc/types_chain.md [hunk 16]
```diff
@@ -874,6 +1490,210 @@ Human-readable duration.
...
+
+## TransactionScheduling {#transactionscheduling}
+
+The scheduling mode of a transaction.
+
+One of:
+
+* `Standard` No special scheduling applied.
+
+* `FutureEra` Execution should be scheduled fro the specific era.
```
Comment (ipopescu) [inline source/docs/casper/developers/json-rpc/types_chain.md:1585]:
```suggestion
* `FutureEra` Execution should be scheduled for the specific era.
```

File: source/docs/casper/developers/json-rpc/types_chain.md [hunk 17]
```diff
@@ -874,6 +1490,210 @@ Human-readable duration.
...
+
+* `stored` The execution target is a stored entity or package.
+
+    Required parameters for a `stored` target:
+
+    * [`id`](#transactioninvocationtarget)
+
+    * [`runtime`](#transactionruntime)
+
+* `session` The exectuion target is the included module bytes.
```
Comment (ipopescu) [inline source/docs/casper/developers/json-rpc/types_chain.md:1627]:
```suggestion
* `session` The execution target is the included module bytes.
```

File: source/docs/casper/developers/json-rpc/types_chain.md [hunk 18]
```diff
@@ -932,6 +1768,8 @@ The actual transformation performed while executing a Deploy.
 
 * `AddKeys` Adds the given collection of [named keys](#namedkey).
 
+* `Prune` Removes the pathing to the global state entry of the specified key. The pruned element remains reachable from previously generated global state root hashes, but will not be included in the next generated global state root hash and subsequent states.
+
 * `Failure` A failed transformation, containing an error message.
```
Comment (ipopescu) [inline source/docs/casper/developers/json-rpc/types_chain.md:1773]:
```suggestion
* `Failure` A failed transformation containing an error message.
```

**Review by ipopescu — APPROVED** [review]
Fantastic content!

I left a few minor suggestions.

File: source/docs/casper/developers/json-rpc/types_chain.md (hunk 2 above)
Comment (ACStoneCL) [reply source/docs/casper/developers/json-rpc/types_chain.md:679]: Yeah, that might be for the best. After I fold in these suggestions, I'll add another commit to the V1 types.

**Pull request merged**